testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/LoginPage.java

```diff
@@ -124,8 +128,20 @@ public void cancel() {
         cancelButton.click();
     }
 
+    public String getInputError() {
+        try {
+            return inputError != null ? inputError.getText() : null;
+        } catch (NoSuchElementException e) {
+            return null;
+        }
+    }
+
     public String getError() {
-        return loginErrorMessage != null ? loginErrorMessage.getText() : null;
+        try {
+            return loginErrorMessage != null ? loginErrorMessage.getText() : getInputError();
```

Comment:
The test should be deterministic and should know whether it's fetching the error message from the notification or from the field.